src/main/scala/akka/persistence/cassandra/journal/StreamMergerActor.scala

```diff
@@ -112,6 +119,8 @@ private class StreamMergerActor(
           case MergeFailure(_, _, _, _) =>
             merging(journalIdProgress, persistenceIdProgress, step + 50l)
           case MergeSuccess(newJournalIdProgress, newPersistenceIdProgress, mergedStream) =>
+            writeIndexProgress(mergedStream)
+            writeProgress(newJournalIdProgress, newPersistenceIdProgress)
```

Comment:
These are actually asynchronous write operations. Shouldn't they be sequenced in a for-comprehension? This way we first write the index and only if that succeeds we write the progress. That's one round trip more but we have never false positives (i.e. a written progress for which the index write failed)